src/messaging/ExchangeMgr.cpp

```diff
@@ -232,7 +231,7 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
             }
 
             // Matched ExchangeContext; send to message handler.
-            ec.HandleMessage(packetHeader, payloadHeader, std::move(msgBuf));
+            ec.HandleMessage(packetHeader, payloadHeader, source, std::move(msgBuf));
```

Comment:
Same, ExchangeContext should only handle Node address instead of low level IP address